openc3/spec/models/target_model_spec.rb

```diff
@@ -67,12 +108,87 @@ module OpenC3
         model = TargetModel.new(folder_name: "SPEC", name: "SPEC", scope: "DEFAULT")
         model.create
         all = TargetModel.all(scope: "DEFAULT")
+        expect(all).to_not be_nil
         expect(all.keys).to contain_exactly("TEST", "SPEC")
       end
     end
 
+    describe "render" do
+      it "renders" do
+        template = '_template.erb'
+        model = TargetModel.new(folder_name: "INST", name: "INST", scope: "DEFAULT")
+        model.create
+        Dir.mktmpdir do |tmpdir|
+          tf = File.open(File.join(tmpdir, template), 'w')
+          tf.puts "CMD_LOG_CYCLE_TIME 1"
+          tf.close
+          model.render(File.expand_path(tf.path), {opt1: '1', opt2: '2', opt3: '3'})
+          # where does the rendered value go?
+        end
+      end
+    end
+
     # self.all_modified & self.download aren't unit tested because it's basically just mocking the entire S3 API
 
+    describe "self.all_modified" do
+      it "returns all the modified targets" do
+        model = TargetModel.new(folder_name: "INST", name: "INST", scope: "DEFAULT")
+        model.create
+        model = TargetModel.new(folder_name: "SPEC", name: "SPEC", scope: "DEFAULT")
+        model.create
+        all_targs = TargetModel.all_modified(scope: "DEFAULT")
+        expect(all_targs.keys).to contain_exactly("INST", "SPEC")
+      end
+    end
+
+    describe "self.modified_files" do
+      it "returns all the modified files" do
+        model = TargetModel.new(folder_name: "TEST", name: "TEST", scope: "DEFAULT")
+        model.create
+        model = TargetModel.new(folder_name: "SPEC", name: "SPEC", scope: "DEFAULT")
+        model.create
+        mods = TargetModel.modified_files('TEST', scope: "DEFAULT")
+        expect(mods).to match_array([])
+      end
+    end
+
+    describe "self.delete_modified" do
+      it "returns all the deleted or modified whatnots" do
+        model = TargetModel.new(folder_name: "TEST", name: "TEST", scope: "DEFAULT")
+        model.create
+        model = TargetModel.new(folder_name: "SPEC", name: "SPEC", scope: "DEFAULT")
+        model.create
+        dels = TargetModel.delete_modified('TEST', scope: "DEFAULT")
+=begin
+Oh, joy.
+        puts 'OpenStruct key="blah"'.encoding.to_s
+        puts 'OpenStruct key="blah"'.bytes.to_s
+
+        puts dels[0].to_s.encoding.to_s
+        puts dels[0].to_s.bytes.to_s
+
+ASCII-8BIT
+[79, 112, 101, 110, 83, 116, 114, 117, 99, 116, 32, 107, 101, 121, 61, 34, 98, 108, 97, 104, 34]
+
+UTF-8
+[35, 60, 79, 112, 101, 110, 83, 116, 114, 117, 99, 116, 32, 107, 101, 121, 61, 34, 98, 108, 97, 104, 34, 62]
+=end
+
+        #expect(dels[0]).to match(/<OpenStruct key="blah">/) #match(/Error deleting object bucket/)
```

Comment:
This isn't testing anything ... there are no expect statements

Reply:
Done.  An oversight: it should already have been removed.